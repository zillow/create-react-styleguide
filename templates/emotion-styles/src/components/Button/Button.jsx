import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

export const StyledButton = styled('button')(props => props.theme.Button);
const Button = props => <StyledButton {...props} />;

Button.propTypes = {
    /** Button content */
    children: PropTypes.node.isRequired,
    /** Button type */
    type: PropTypes.oneOf(['primary', 'secondary']),
    /** Enable outline buttons */
    outline: PropTypes.bool,
};

Button.defaultProps = {
    type: 'primary',
    outline: false,
};

export default Button;
