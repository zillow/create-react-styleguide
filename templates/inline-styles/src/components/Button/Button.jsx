import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Button extends React.Component {
    static propTypes = {
        /** Button content */
        children: PropTypes.node.isRequired,
        /** Button type */
        type: PropTypes.oneOf(['primary', 'secondary']),
        /** Enable outline buttons */
        outline: PropTypes.bool
    };
    static defaultProps = {
        type: 'primary'
    };

    render() {
        let styles = {
            display: 'inline-block',
            fontWeight: '400',
            textAlign: 'center',
            whiteSpace: 'nowrap',
            verticalAlign: 'middle',
            userSelect: 'none',
            border: '1px solid transparent',
            padding: '.375rem .75rem',
            fontSize: '1rem',
            lineHeight: '1.5',
            borderRadius: '.25rem',
            transition: 'color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out',
            color: '#fff',
            backgroundColor: '#007bff',
            borderColor: '#007bff'
        };

        if (this.props.outline && this.props.type == 'primary') {
            styles = Object.assign(styles, {
                color: '#007bff',
                backgroundColor: 'transparent',
                backgroundImage: 'none',
                borderColor: '#007bff'
            });
        } else if (this.props.outline && this.props.type == 'secondary') {
            styles = Object.assign(styles, {
                color: '#6c757d',
                backgroundColor: 'transparent',
                backgroundImage: 'none',
                borderColor: '#6c757d'
            });
        } else if (this.props.type === 'secondary') {
            styles = Object.assign(styles, {
                backgroundColor: '#6c757d',
                borderColor: '#6c757d'
            });
        }

        return <button style={styles} {...this.props} />;
    }
}
