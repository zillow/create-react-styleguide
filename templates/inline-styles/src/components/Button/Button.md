Button variants:

```jsx
<div>
    <Button>Primary</Button> <Button variant="secondary">Secondary</Button>
</div>
```

Outline buttons:

```jsx
<div>
    <Button outline={true}>Primary</Button> <Button outline variant="secondary">Secondary</Button>
</div>
```
