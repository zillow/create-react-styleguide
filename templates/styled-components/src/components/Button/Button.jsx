import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ButtonMixin } from '../../mixins/ButtonMixin';

export const StyledButton = styled.button`
    ${ButtonMixin}
`;

/**
 * A simple wrapper around a native
 * [`button`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) element.
 */
const Button = React.forwardRef((props, ref) => <StyledButton ref={ref} {...props} />);

Button.displayName = 'Button';

Button.propTypes = {
    /** Button content */
    children: PropTypes.node.isRequired,
    /** Button type */
    variant: PropTypes.oneOf(['primary', 'secondary']),
    /** Enable outline buttons */
    outline: PropTypes.bool,
};

Button.defaultProps = {
    variant: 'primary',
    outline: false,
};

export default Button;
