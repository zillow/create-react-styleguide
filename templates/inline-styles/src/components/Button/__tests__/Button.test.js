import React from 'react';
import TestRenderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
    it('renders a primary button', () => {
        const testRenderer = TestRenderer.create(<Button>Click Me!</Button>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('renders a primary button with outline', () => {
        const testRenderer = TestRenderer.create(<Button outline>Click Me!</Button>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('renders a secondary button', () => {
        const testRenderer = TestRenderer.create(<Button variant="secondary">Click Me!</Button>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('renders a secondary button with outline', () => {
        const testRenderer = TestRenderer.create(
            <Button variant="secondary" outline>
                Click Me!
            </Button>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
});
