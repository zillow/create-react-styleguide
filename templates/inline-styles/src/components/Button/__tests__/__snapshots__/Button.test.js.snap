// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a primary button 1`] = `
<button
  style={
    Object {
      "backgroundColor": "#007bff",
      "border": "1px solid transparent",
      "borderColor": "#007bff",
      "borderRadius": ".25rem",
      "color": "#fff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "1rem",
      "fontWeight": "400",
      "lineHeight": "1.5",
      "padding": ".375rem .75rem",
      "textAlign": "center",
      "userSelect": "none",
      "verticalAlign": "middle",
      "whiteSpace": "nowrap",
    }
  }
  type="button"
>
  Click Me!
</button>
`;

exports[`Button renders a primary button with outline 1`] = `
<button
  style={
    Object {
      "backgroundColor": "transparent",
      "backgroundImage": "none",
      "border": "1px solid transparent",
      "borderColor": "#007bff",
      "borderRadius": ".25rem",
      "color": "#007bff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "1rem",
      "fontWeight": "400",
      "lineHeight": "1.5",
      "padding": ".375rem .75rem",
      "textAlign": "center",
      "userSelect": "none",
      "verticalAlign": "middle",
      "whiteSpace": "nowrap",
    }
  }
  type="button"
>
  Click Me!
</button>
`;

exports[`Button renders a secondary button 1`] = `
<button
  style={
    Object {
      "backgroundColor": "#6c757d",
      "border": "1px solid transparent",
      "borderColor": "#6c757d",
      "borderRadius": ".25rem",
      "color": "#fff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "1rem",
      "fontWeight": "400",
      "lineHeight": "1.5",
      "padding": ".375rem .75rem",
      "textAlign": "center",
      "userSelect": "none",
      "verticalAlign": "middle",
      "whiteSpace": "nowrap",
    }
  }
  type="button"
>
  Click Me!
</button>
`;

exports[`Button renders a secondary button with outline 1`] = `
<button
  style={
    Object {
      "backgroundColor": "transparent",
      "backgroundImage": "none",
      "border": "1px solid transparent",
      "borderColor": "#6c757d",
      "borderRadius": ".25rem",
      "color": "#6c757d",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "1rem",
      "fontWeight": "400",
      "lineHeight": "1.5",
      "padding": ".375rem .75rem",
      "textAlign": "center",
      "userSelect": "none",
      "verticalAlign": "middle",
      "whiteSpace": "nowrap",
    }
  }
  type="button"
>
  Click Me!
</button>
`;
